### PR TITLE
Add Marvel Snap Provider

### PR DESCRIPTION
**Pull request by isvora**
Added provider for the new trading card game called Marvel Snap

### CODE REVIEW

File: src/main/java/net/datafaker/service/files/EnFile.java [hunk 1]
```diff
@@ -147,6 +147,7 @@ public String getPath() {
         "lorem.yml",
 //            "lovecraft.yml",
 //            "markdown.yml",
+        "marvel_snap",
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/service/files/EnFile.java:150]: why not yml?

File: src/main/java/net/datafaker/providers/videogame/VideoGameProviders.java [hunk 2]
```diff
@@ -42,6 +42,9 @@ default LeagueOfLegends leagueOfLegends() {
     default MassEffect massEffect() {
         return getProvider(MassEffect.class, MassEffect::new);
     }
+    default MarvelSnap marvelSnap() {
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/providers/videogame/VideoGameProviders.java:45]: Please keep abc order

File: README.md [hunk 3]
```diff
@@ -357,6 +357,7 @@ Providers
 * Lord Of The Rings
 * Locality
 * Lorem
+* Marvel Snap
```
Comment (snuyanzin) [inline README.md:360]: Please keep abc order

File: src/main/java/net/datafaker/providers/videogame/MarvelSnap.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import net.datafaker.providers.base.AbstractProvider;
+
+public class MarvelSnap extends AbstractProvider<VideoGameProviders> {
+
+    protected MarvelSnap(final VideoGameProviders faker) {
+        super(faker);
+    }
+
+    public String characters() {
```
Comment (ghusta) [inline src/main/java/net/datafaker/providers/videogame/MarvelSnap.java:11]: I think it's better to use the singular form for the getter, as it returns a single element.
For example : 
```
String character = faker.marvelSnap().character();
```

Comment (isvora) [reply src/main/java/net/datafaker/providers/videogame/MarvelSnap.java:11]: Sure thing.

File: src/main/java/net/datafaker/service/files/EnFile.java (hunk 1 above)
Comment (isvora) [reply src/main/java/net/datafaker/service/files/EnFile.java:150]: oops, I forgot it

File: src/main/java/net/datafaker/providers/videogame/MarvelSnap.java (hunk 4 above)
Comment (bodiam) [reply src/main/java/net/datafaker/providers/videogame/MarvelSnap.java:11]: This doesn't look too be resolved yet.

File: src/main/java/net/datafaker/providers/videogame/MarvelSnap.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        return faker.marvelSnap().characters();
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/providers/videogame/MarvelSnap.java:12]: looks like endless recursion

**Comment by bodiam** [comment]
Nice, thanks! Could you also please make the other methods singular? Thanks!

File: src/main/java/net/datafaker/service/files/EnFile.java [hunk 6]
```diff
@@ -147,6 +147,7 @@ public String getPath() {
         "lorem.yml",
 //            "lovecraft.yml",
 //            "markdown.yml",
+        "marvel_snap.yml",
```
Comment (bodiam) [inline src/main/java/net/datafaker/service/files/EnFile.java:150]: Why is marvel_snap added here? I don't think that's needed.

File: src/test/java/net/datafaker/providers/videogame/MarvelSnapTest.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    @Test
+    void charactersTest() {
+        String character = faker.marvelSnap().character();
+        assertThat(character).matches("[ A-Za-z']+");
+    }
+
+    @Test
+    void zonesTest() {
+        String zone = faker.marvelSnap().zones();
+        assertThat(zone).matches("[ A-Za-z]+");
```
Comment (bodiam) [inline src/test/java/net/datafaker/providers/videogame/MarvelSnapTest.java:18]: this regular expression is incorrect. I doesn't accommodate for Kamar-Taj. Can you please change the `@Test` annotations to `@RepeatedTest(10)`. That has a far higher chance of spotting these issues, and it's fast enough.

File: src/main/java/net/datafaker/service/files/EnFile.java (hunk 6 above)
Comment (isvora) [reply src/main/java/net/datafaker/service/files/EnFile.java:150]: correct, I'll remove it.

**Comment by bodiam** [comment]
Thanks @isvora , looks good to me!

**Pull request merged**